manager/orchestrator/taskreaper/task_reaper.go

```diff
@@ -94,16 +101,16 @@ func (tr *TaskReaper) Run() {
 }
 
 func (tr *TaskReaper) tick() {
-	if len(tr.dirty) == 0 {
+	if len(tr.dirty) == 0 && len(tr.orphaned) == 0 {
 		return
 	}
 
 	defer func() {
 		tr.dirty = make(map[instanceTuple]struct{})
+		tr.orphaned = nil
 	}()
 
-	var deleteTasks []string
-
+	deleteTasks := tr.orphaned
```

Comment:
If orphaned tasks get deleted right away, why is the `ORPHANED` state useful? Why can't the dispatcher just delete the tasks directly?